### PR TITLE
add registeredIn search scope

### PR DESCRIPTION
**Pull request by tareq89**
## Description
[Issue](https://github.com/opencrvs/opencrvs-core/issues/12611)
[core pr](https://github.com/opencrvs/opencrvs-core/pull/12624)
[countryconfig pr](https://github.com/opencrvs/opencrvs-countryconfig/pull/1424)
[farajaland pr](https://github.com/opencrvs/opencrvs-farajaland/pull/2150)

Clearly describe what has been changed. Include relevant context or background.
Explain how the issue was fixed (if applicable) and the root cause.

Link this pull request to the GitHub issue (and optionally name the branch `ocrvs-<issue #>`)

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: src/data-seeding/roles/roles.ts
Comment (jamil314) [inline src/data-seeding/roles/roles.ts:1]: This change will not restrict anything, rather allows searching for records with placeOfEvent outside admin area but registered within admin area

Comment (tareq89) [reply src/data-seeding/roles/roles.ts:1]: I think that's okay and intentional. Because if we go for the AND logic, most of the E2E will be broken. Tried it. And also, even if the placeOfEvent is outside of the jurisdiction, a user should be able to interact with the event if it is registered in his office, shouldn't he?

**Review by makelicious — CHANGES_REQUESTED** [review]
options are combined using AND withi scope. With these changes, user will only be able to search for events that took place in their administrative area, and were registered there.

File: e2e/testcases/advanced-search/8-search-death-location.spec.ts [hunk 1]
```diff
@@ -34,8 +34,8 @@ test.describe("Advanced Search 8 - Death - Deceased's place of death", () => {
 
     await expect(page.getByText('Country')).not.toBeVisible()
 
-    await page.locator('#eventDetails____deathLocation').fill('Zobwe')
```
Comment (makelicious) [inline e2e/testcases/advanced-search/8-search-death-location.spec.ts:37]: Could you write a test case. This does not test the change itself.

**Review by makelicious — CHANGES_REQUESTED** [review]

**Comment by tareq89** [comment]
> options are combined using AND withi scope. With these changes, user will only be able to search for events that took place in their administrative area, and were registered there.

Yes, we can decide that, but earlier I have tried with AND and it turned out 90% e2e shards started failing. Because, when e2es were written, Jurisdiction wasn't a thing.

File: src/data-seeding/roles/roles.ts
Comment (jamil314) [reply src/data-seeding/roles/roles.ts:1]: Can the user search for a record where placeOfEvent is inside their jurisdiction but it was registered outside of their jurisdiction?

Comment (makelicious) [reply src/data-seeding/roles/roles.ts:1]: jamil has a point. If we can't do the same as for the form, and move 'allowedLocations' property something for countries to configure.

We would need to define both of them to keep the original functionality, and satisfy the need for additional property.
```
{ type: 'record.search', options: { placeOfEvent: 'administrativeArea' } }, // keeps the original behaviour
{ type: 'record.search', options: { placeOfEvent: 'administrativeArea', registeredIn: 'administrativeArea' } }, // satisfies your need for registeredIn property to be defined
```

Comment (makelicious) [reply src/data-seeding/roles/roles.ts:1]: https://github.com/opencrvs/e2e/actions/runs/25860443165/job/75989548817

ah nice, we seem to have a test case just for that?
